core/deployment/src/main/java/io/quarkus/deployment/builditem/ContainerRuntimeStatusBuildItem.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.deployment.builditem;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.deployment.IsContainerRuntimeWorking;
+
+public abstract class ContainerRuntimeStatusBuildItem extends SimpleBuildItem {
+    private final IsContainerRuntimeWorking isContainerRuntimeWorking;
+    private Boolean cachedStatus;
+
+    protected ContainerRuntimeStatusBuildItem(IsContainerRuntimeWorking isContainerRuntimeWorking) {
+        this.isContainerRuntimeWorking = isContainerRuntimeWorking;
+    }
+
+    public boolean isContainerRuntimeAvailable() {
+        synchronized (this) {
+            if (cachedStatus == null) {
+                cachedStatus = isContainerRuntimeWorking.getAsBoolean();
+            }
+        }
```

Comment:
Not too important, but this could be improved like so:

```suggestion
        if (cachedStatus == null) {
          synchronized (this) {
            if (cachedStatus == null) {
                cachedStatus = isContainerRuntimeWorking.getAsBoolean();
            }
          }
        }
```

Reply:
Good catch! I'll make that change this morning.

Originally the entire method had the `synchronized` keyword on it, so I think this is a big improvement. No need to synchronize the whole method.